src/objects/Card.ts

```diff
@@ -0,0 +1,136 @@
+import { ICardVisitor } from "../visitor/ICardVisitor";
+import { ICard, ICardCritter, ICardItem } from "./interfaces";
+import { Card as CardPayload } from "../rest/payloads/cards";
+
+/**
+ * @Created 2024-09-28
+ * @Brief this file contains objects for both critter and item cards.
+ */
+
+export abstract class Card implements ICard {
+    private readonly _cardid: number;
+    private readonly _name: string;
+    private readonly _playcost: number;
+    private readonly _rarity: number;
+    private readonly _image: string;
+    private readonly _description: string;
+
+    public static fromCardPayload(payload: CardPayload): ICardCritter | ICardItem {
+        if (payload.type === "critter") {
+            return CardCritter.fromCardPayload(payload);
+        } else if (payload.type === "item") {
+            return CardItem.fromCardPayload(payload);
+        } else{
+            throw new Error("Card type not recognized.");
+        }
+    }
+
+    constructor(cardid: number, name: string, playcost: number, rarity: number, image: string, description: string) {
+        this._cardid = cardid;
+        this._name = name;
+        this._playcost = playcost;
+        this._rarity = rarity;
+        this._image = image;
+        this._description = description;
+    }
+
+    public get cardid(): number {
+        return this._cardid;
+    }
+
+    public get name(): string {
+        return this._name;
+    }
+
+    public get playcost(): number {
+        return this._playcost;
+    }
+
+    public get rarity(): number {
+        return this._rarity;
+    }
+
+    public get image(): string {
+        return this._image;
+    }
+
+    public get description(): string {
+        return this._description;
+    }
+
+    accept(visitor: ICardVisitor): void {
+        throw new Error("Method not implemented: this card should either be a critter or an item.");
+    }
+}
+
+export class CardCritter extends Card implements ICardCritter {
+    private readonly _damage: number;
+    private readonly _health: number;
+    private readonly _abilities: number[];
+
+    public static fromCardPayload(payload: CardPayload): ICardCritter {
+        return new CardCritter(
+            payload.cardid,
+            payload.name,
+            payload.playcost,
+            payload.rarity,
+            payload.image,
+            payload.description,
+            (payload.type_specific as { damage: number }).damage,
+            (payload.type_specific as { health: number }).health,
+            (payload.type_specific as { abilities: number[] }).abilities
```

Comment:
maybe cast this once to an intermediate object, or look into whether we could parameterize CardPayload so we'd have CardPayload<CardCritter> and CardPayload<CardItem> not sure if thats worth the effort though tbh

Reply:
should user the CardCritter type defined here: https://github.com/InternetEnemies/combatcritters-ts/blob/main/src/rest/payloads/cards.ts

Reply:
> should user the CardCritter type defined here: https://github.com/InternetEnemies/combatcritters-ts/blob/main/src/rest/payloads/cards.ts

updated